### PR TITLE
fix: Error while adding purchase invoice and purchase receipt in Assets

### PR DESCRIPTION
**Pull request by hitenshKharva**
Ref : [Asana Task](https://app.asana.com/0/1199082161983365/1199694816340111)

https://user-images.githubusercontent.com/56474017/103635831-16f43780-4f6f-11eb-89a2-88780b1ebb82.mp4

### CODE REVIEW

**Review by vishdha — DISMISSED** [review]

File: erpnext/controllers/queries.py [hunk 1]
```diff
@@ -550,50 +550,32 @@ def item_manufacturer_query(doctype, txt, searchfield, start, page_len, filters)
 @frappe.whitelist()
 def get_purchase_receipts(doctype, txt, searchfield, start, page_len, filters):
 	fields = get_fields("Purchase Receipt", ["name"])
+	item_filters = [
+		['Purchase Receipt', 'docstatus', '=', '1'],
+		['Purchase Receipt Item','name', 'like', '%' + txt + '%']
```
Comment (hrwX) [inline erpnext/controllers/queries.py]:
```suggestion
		['Purchase Receipt Item','name', 'like', frappe.db.escape('%{0}%'.format(txt))]
```

File: erpnext/controllers/queries.py [hunk 2]
```diff
@@ -550,50 +550,32 @@ def item_manufacturer_query(doctype, txt, searchfield, start, page_len, filters)
...
-		})
-
-	return frappe.db.sql(query, flt)
-
+	purchase_receipts = frappe.get_all('Purchase Receipt',
+		fields=fields,
+		filters=item_filters,
+		as_list=1
+	)
+	return purchase_receipts
```
Comment (hrwX) [inline erpnext/controllers/queries.py:564]: Where is the `item_code` condition ?

**Review by hrwX — CHANGES_REQUESTED** [review]

File: erpnext/controllers/queries.py (hunk 2 above)
Comment (hitenshKharva) [reply erpnext/controllers/queries.py:564]: added all the required conditions.

Comment (vishdha) [reply erpnext/controllers/queries.py:564]: Item code filter comes from queries.js where it filters at line 555

**Review by vishdha — DISMISSED** [review]

File: erpnext/controllers/queries.py [hunk 3]
```diff
@@ -550,50 +550,32 @@ def item_manufacturer_query(doctype, txt, searchfield, start, page_len, filters)
 @frappe.whitelist()
 def get_purchase_receipts(doctype, txt, searchfield, start, page_len, filters):
 	fields = get_fields("Purchase Receipt", ["name"])
+	item_filters = [
+		['Purchase Receipt', 'docstatus', '=', '1', 'and', 'name', 'like', frappe.db.escape('%{0}%'.format(txt))],	
+		['Purchase Receipt Item', 'item_code', '=', filters.get("item_code"), 'and', 'parent', '=', fields[0]]
+	]
```
Comment (hrwX) [inline erpnext/controllers/queries.py]: What is this query ?
```suggestion
	item_filters = [
		['Purchase Receipt', 'docstatus', '=', '1'],
		['Purchase Receipt', 'name', 'like', frappe.db.escape('%{0}%'.format(txt))],	
		['Purchase Receipt Item', 'item_code', '=', filters.get("item_code")],
		['Purchase Receipt Item', 'parent', '=', fields[0]]
	]
```

File: erpnext/controllers/queries.py [hunk 4]
```diff
@@ -550,50 +550,32 @@ def item_manufacturer_query(doctype, txt, searchfield, start, page_len, filters)
...
-		flt.append({
-			"item_code": frappe.db.escape(filters.get('item_code'))
-		})
-
-	return frappe.db.sql(query, flt)
+	fields = get_fields("Purchase Invoice", ["name"])
+	item_filters =[
+		['Purchase Invoice', 'docstatus', '=', '1', 'and', 'name', 'like', frappe.db.escape('%{0}%'.format(txt))],
+		['Purchase Invoice Item', 'item_code', '=', filters.get("item_code"), 'and', 'parent', '=', fields[0]]
+	]
```
Comment (hrwX) [inline erpnext/controllers/queries.py]:
```suggestion
	item_filters =[
		['Purchase Invoice', 'docstatus', '=', '1'],
		['Purchase Invoice', 'name', 'like', frappe.db.escape('%{0}%'.format(txt))],
		['Purchase Invoice Item', 'item_code', '=', filters.get("item_code")],
		['Purchase Invoice Item', 'parent', '=', fields[0]]
	]
```

**Review by hrwX — CHANGES_REQUESTED** [review]

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**